pkg/jobstore/boltdb/index.go

```diff
@@ -18,50 +36,35 @@ func NewIndex(bucketPath string) *Index {
 	}
 }
 
-func (i *Index) Add(tx *bolt.Tx, label []byte, identifier []byte) error {
-	bkt, err := i.rootBucketPath.Get(tx, true)
+func (i *Index) Add(tx *bolt.Tx, identifier []byte, subpath ...[]byte) error {
+	bkt, err := i.rootBucketPath.Sub(subpath...).Get(tx, true)
 	if err != nil {
 		return err
 	}
 
-	bktLabel, err := bkt.CreateBucketIfNotExists(label)
-	if err != nil {
-		return err
-	}
-
-	_, err = bktLabel.CreateBucketIfNotExists(identifier)
+	_, err = bkt.CreateBucketIfNotExists(identifier)
```

Comment:
I thought I left this comment in my previous review, but I can't find it! In the index, we are creating a bucket for each index value/identifier instead of having a single bucket and each identifier is a key (with empty value) in that bucket.

I don't know how boltdb works internally, but my assumption is that buckets are more expensive to create and query than key/value entries in existing buckets. If this assumption is true, then we we need to revisit how indexes are implemented, and minimize our use of dynamic buckets in other parts of the jobstore.

I understand I suggested exploring nested buckets with a parent `jobs` bucket and have executions, spec and state as child buckets, but please push back if that is not an efficient way to implement boltdb schemas

Reply:
Yes, you're absolutely right, I thought Buckets were really cheap to create, but turns out each one is a newly create btree and so there's a lot of looking up of the location of other btrees that seems unnecessary.  Luckily fixing this was pretty straight-forward in `index.go` so I've switched it to writing 0 byte values.  The only downside of this approach I've seen is that there's a temptation to have a single bucket and prefix the keys, but then we trade a lot of space for the single occassional lookup, so I've left it with a btree per index for now. Impl in fe8f5b7fe8f5b772d01951392e63b8abbcc5df826d1d2a8